GoMeet/src/main/webapp/js/interactive-map.js

```diff
@@ -131,3 +131,37 @@ function buildInfoWindowVote(title, voteCount, note, keyString) {
       document.createElement('br'), button);
   return containerDiv;
 }
+
+/** Creates a list element that represents a location. */
+function createPopularLocationElement(location) {
+  const liElement = document.createElement('li');
+  liElement.className = 'location';
+
+  let titleContainer = document.createElement('span');
+  titleContainer.setAttribute('id', 'popularLocationTitle');
+  titleContainer.innerText = location.title;
+
+  let voteCountContianer = document.createElement('span');
+  voteCountContianer.setAttribute('id', 'popularLocationVoteCount');
+  voteCountContianer.innerText = location.voteCount;
+
+  liElement.append('Title: ', titleContainer, document.createElement('br'),
+      'Number of Votes: ', voteCountContianer);
+  return liElement;
+}
+
+/** Fetches the popular location data and adds it to the DOM. */
+async function displayPopularLocations() {
+  const popularLocationElement = 
+      document.getElementById('popular-locations-container');
+  popularLocationElement.innerHTML = '';
+
+  let json = await MeetingLocationDAO.fetchPopularLocations();
```

Comment:
What happens if this fails? Do we need to handle that kind of error?

Reply:
Yep, I've added that in commit:
- [https://github.com/googleinterns/step271-2020/pull/44/commits/4e9caeb3fd9fdf82305862ad861e8194cd47c210](url)
- [https://github.com/googleinterns/step271-2020/pull/44/commits/14d350389b1301794404059afba51cd7bc0726ff](url).

What I did was make the DAO throw an error if the locations could not be fetched. The function that calls this in ```interactive-map.js``` would catch the error and pass it to ```handleError()```, which makes an alert. I chose to use a seperate function to handle errors if in the future I want to change how errors are displayed. 

I will do the same for fetchLocations() but I thought it didn't belong in this PR. So, I will include it when I refactor the Dao :).

Reply:
SG, thanks for the explanation :)